source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp

```diff
@@ -258,12 +273,15 @@ ShaderPtr GlslFragmentGenerator::generate(const string& fragmentName, ElementPtr
         for (size_t i = 0; i < vertexData.size(); ++i)
         {
             const string& name = vertexData[i]->getVariable();
-
             emitLineBegin(pixelStage);
             emitString(HW::T_VERTEX_DATA_INSTANCE, pixelStage);
             emitString(".", pixelStage);
             emitString(name, pixelStage);
             emitString(" = ", pixelStage);
+            if (OGS_IS_REQUIREMENT_ONLY.find(vertexData[i]->getName()) != OGS_IS_REQUIREMENT_ONLY.end())
+            {
+                emitString("PIX_IN.", pixelStage);
```

Comment:
Will this work with HLSL conversion?

Reply:
Unfortunately, it won't - see the explanation here: https://github.com/autodesk-forks/MaterialX/blob/a29d168ef03113f5ddbd25761f0ba964e83bf415/source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp#L86 Even for GLSL, `PIX_IN` seems to be an undocumented VP2 implementation detail which plugins shouldn't rely on.
So basically `isRequirementOnly` channels can't be accessed in HLSL. `varyingInputParam` OTOH just works: it makes sure that VP2 passes the vertex stream as an argument to the fragment's root function. I see no good use case for `isRequirementOnly` (or am I missing something?) so I would rather refactor `OgsXmlGenerator` to use `varyingInputParam` for all channels and not change anything is this file.